### PR TITLE
Ignore patch version bumps in eth client supported checks

### PR DESCRIPTION
**Pull request by LefterisJP**
Fixes: #4608 
## Description

Check description of issue.

## PR review check list

Quality check list that cannot be automatically verified.

- Safety
    - [ ] The changes respect the necessary conditions for safety (https://raiden-network-specification.readthedocs.io/en/latest/smart_contracts.html#protocol-values-constraints)
-  Code quality
    - [ ] Error conditions are handled
    - [ ] Exceptions are propagated to the correct parent greenlet
    - [ ] Exceptions are correctly classified as recoverable or unrecoverable
- Compatibility
    - [ ] State changes are forward compatible
    - [ ] Transport messages are backwards and forward compatible
- Commits
    - [ ] Have good messages
    - [ ] Squashed unecessary commits
- If it's a bug fix:
    - Regression test for the bug
        - [ ] Properly covers the bug
        - [ ] If an integration test is used, it could not be written as a unit test
- Is it a user facing feature/bug fix?
    - [ ] Changelog entry has been added

### CODE REVIEW

File: raiden/utils/ethereum_clients.py [hunk 1]
```diff
@@ -24,6 +25,32 @@ def parse_geth_version(client_version: str) -> Optional[tuple]:
     return parse_version(matches.groups()[0])
 
 
+def support_check(
```
Comment (palango) [inline raiden/utils/ethereum_clients.py]: Would be nice to have unit tests for this function.

Besides that: where did you find the `Version`  class? I cannot find it.

File: raiden/utils/ethereum_clients.py [hunk 2]
```diff
@@ -24,6 +25,32 @@ def parse_geth_version(client_version: str) -> Optional[tuple]:
...
+    our_version: Version,
+    highest_supported_version_string: str,
+    lowest_supported_version_string: str,
+) -> bool:
+    if len(our_version._key[1]) != 3:
+        # This can happen for versions such as 0.0.0
+        return False
+
+    # our_major_num = our_version._key[1][0]
```
Comment (palango) [inline raiden/utils/ethereum_clients.py]: Are these comments still necessary?

Comment (LefterisJP) [reply raiden/utils/ethereum_clients.py]: Definitely not .... it's left over commented code, thank you for spotting it!

File: raiden/utils/ethereum_clients.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/utils/ethereum_clients.py]: It's tested via the unit tests in test_cli.py (https://github.com/raiden-network/raiden/pull/4612/files#diff-4f31d096585cf25811afbaf6fcc4fb7c)

> Besides that: where did you find the Version class? I cannot find it.

I went in the virtual env dependency of pkg_resources and found it in there. I wanted to see what does `parse_version` return. For the full path check how I import it in that file.

**Pull request merged**